m0t1fs/linux_kernel/file.c

```diff
@@ -5935,14 +5940,14 @@ static void cc_bottom_half(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	if (csb->csb_confc_state.cus_state != M0_CC_READY)
 		rc = M0_ERR(-ESTALE);
 	m0_mutex_unlock(&csb->csb_confc_state.cus_lock);
+
 ref_dec:
 	if (ti->ti_rc == 0 && rc != 0)
 		ti->ti_rc = rc;
 	if (xfer->nxr_rc == 0 && rc != 0)
 		xfer->nxr_rc = rc;
 	m0_fop_put0_lock(&cc_fop->crf_fop);
-	if (reply_fop != NULL)
```

Comment:
Why we are removing NULL check for reply_fop here ?
are we sure that reply_fop will never be NULL at this place.

Reply:
The m0_fop_put_lock() will assume the argument 'fop' is not NULL.
The m0_fop_put0_lock() will check if the argument 'fop' is NULL or not.

So, the the checking here is redundant.